lib/std/js/jsxmlhttprequest.nim

```diff
@@ -0,0 +1,57 @@
+## `XMLHttpRequest` for the JavaScript target: https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest
+when not defined(js):
+  {.fatal: "Module jsxmlhttprequest is designed to be used with the JavaScript backend.".}
+
+from std/dom import Node
+
+type XMLHttpRequest* = ref object of JsRoot  ## https://xhr.spec.whatwg.org
+  responseXML*: Node
+  withCredentials*: bool
+  status*, timeout*, readyState*: cint
+  responseText*, responseURL*, statusText*: cstring
+
+func newXMLHttpRequest*(): XMLHttpRequest {.importjs: "new XMLHttpRequest()".}
+  ## Constructor for `XMLHttpRequest`.
+
+func open*(this: XMLHttpRequest; metod, url: cstring; async = true; user = cstring.default; password = cstring.default) {.importjs: "#.$1(#, #, #, #, #)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/open
+
+func send*(this: XMLHttpRequest; body: cstring | Node = cstring.default) {.importjs: "#.$1(#)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/send
+
+func abort*(this: XMLHttpRequest) {.importjs: "#.$1()".}
+  ## https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/abort
+
+func getAllResponseHeaders*(this: XMLHttpRequest): cstring {.importjs: "#.$1()".}
+  ## https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/getAllResponseHeaders
+
+func overrideMimeType*(this: XMLHttpRequest; mimeType: cstring) {.importjs: "#.$1(#)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/overrideMimeType
+
+func setRequestHeader*(this: XMLHttpRequest; key, value: cstring) {.importjs: "#.$1(#, #)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/setRequestHeader
+
+func setRequestHeader*(this: XMLHttpRequest; keyValuePairs: openArray[tuple[key, val: cstring]]) {.importjs:
+  "(() => { const rqst = #; #.forEach((item) => rqst.$1(item[0], item[1])) })()".}
+  ## Same as `setRequestHeader` but takes `openArray[tuple[key, val: cstring]]`.
+
+
+runnableExamples:
+  from std/dom import Node
+  if defined(fusionJsXmlhttprequestTests):
```

Comment:
`fusionJsXmlhttprequestTests` would be wrong here (stdlib, not fusion), but anyways, now the preferred way is `-r:off` when we can't run things in CI